rootex/core/resource_files/custom_material_resource_file.cpp

```diff
@@ -427,4 +495,104 @@ void CustomMaterialResourceFile::draw()
 
 		ImGui::TreePop();
 	}
+
+	int i = 0;
+	int j = 0;
+	float tempFloat3[3];
+	float* addressesFloat3[3];
+	float tempColor[4];
+	float* addressesColor[4];
+	for (auto&& cb : customConstantBuffers)
+	{
+		if (typeOfCustomConstantBuffers[i / 4] == 1.0)
+		{
+			if (i % 4 == 0)
+			{
+				String customConstantBufferName = "CB Slot " + std::to_string(i / 4);
+				ImGui::DragFloat(customConstantBufferName.c_str(), &cb, 0.01f, 0.0f, 10.0f);
+				ImGui::Separator();
+			}
+		}
+		if (typeOfCustomConstantBuffers[i / 4] == 2.0)
+		{
+			tempFloat3[j % 3] = cb;
+			addressesFloat3[j % 3] = &cb;
+			if (i % 4 != 3)
+			{
+				j++;
+				i++;
+				continue;
+			}
```

Comment:
this code may be logically correct, but logically most appealing soln is often not maintainable/easy to read.
prioritize clarity

Reply:
yeah I think j is unnecessary